std/algorithm/comparison.d

```diff
@@ -1911,3 +1915,101 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
         [mytuple(2, 3), mytuple(1, 2)]
     ));
 }
+
+enum ifTestable(T, alias pred = a => a) = __traits(compiles, { if ((pred(T()))) {} }) || isArray!T;
```

Comment:
I think you want to use `T.init` instead of `T()`. The former works even for types with disabled `this()`. Also, why did you need to special case `isArray!T`? Double also, why are static arrays included? They can't be tested with `if`.